File: 78-subsets/78-subsets.java
class Solution {
    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>> subsets = new ArrayList<>();
        int n = nums.length;
        for (int m = 0; m < 1 << n; m++) {
            List<Integer> subset = new ArrayList<>();
            for (int i = 0; i < n; i++) {
                if ((m & 1 << i) > 0) subset.add(nums[i]);
            }
            subsets.add(subset);
        }
        return subsets;
    }
}

File: 78-subsets/78-subsets.kt
class Solution {
    fun subsets(nums: IntArray): List<List<Int>> {
        val subsets = arrayListOf<List<Int>>()
        for (m in 0 until (1 shl nums.size)) {
            val subset = arrayListOf<Int>()
            for (i in nums.indices) {
                if (m and (1 shl i) > 0) subset.add(nums[i])
            }
            subsets.add(subset)
        }
        return subsets
    }
}